### PR TITLE
CORE-799: Add TransferAttribute

### PR DESCRIPTION
**Pull request by EBGToo**
[Java work covered in CORE-807]  There are things to discuss about this Swift implementation (use of a value type for `TransferAttribute`) that might need to change after discussing Java implementation.  But, looking to get PR in place and willing to modify later.

### CODE REVIEW

File: crypto/BRCryptoTransfer.c [hunk 1]
```diff
@@ -416,6 +420,27 @@ cryptoTransferGetUnitForFee (BRCryptoTransfer transfer) {
     return cryptoUnitTake (transfer->unitForFee);
 }
 
+extern size_t
+cryptoTransferGetAttributeCount (BRCryptoTransfer transfer) {
+    return array_count(transfer->attributes);
```
Comment (michael-brd) [inline crypto/BRCryptoTransfer.c]: There will be thread-sanitizer warnings here.

File: generic/BRGenericRipple.c [hunk 2]
```diff
@@ -253,6 +272,49 @@ genericRippleWalletEstimateFeeBasis (BRGenericWalletRef wallet,
...
+    if (asRequired) { *count = rippleTransactionFieldRequiredCount; return rippleTransactionFieldRequiredNames; }
+    else {            *count = rippleTransactionFieldOptionalCount; return rippleTransactionFieldOptionalNames; }
+}
+
+static int
+genericRippleWalletValidateTransactionAttribute (BRGenericWalletRef wallet,
+                                                 BRGenericTransferAttribute attribute) {
+    if (0 == strcmp (attribute.key, "DestinationTag")) {
+        uint32_t tag;
+        return 1 == sscanf(attribute.value, "%u", &tag);
```
Comment (michael-brd) [inline generic/BRGenericRipple.c]: Might `attribute.value` be NULL?

File: crypto/BRCryptoTransfer.c (hunk 1 above)
Comment (EBGToo) [reply crypto/BRCryptoTransfer.c]: Fixed.  I don't see the BRCryptoWallet additions as needing a lock.

File: generic/BRGenericRipple.c (hunk 2 above)
Comment (EBGToo) [reply generic/BRGenericRipple.c]: Fixed.

**Pull request merged**